src/backend/bs_dodup

```diff
@@ -44,12 +45,47 @@ my $checkinterval_error = 10 * 60;
 $checkinterval_ok = $BSConfig::dodup_checkinterval_ok if $BSConfig::dodup_checkinterval_ok;
 $checkinterval_error = $BSConfig::dodup_checkinterval_error if $BSConfig::dodup_checkinterval_error;
 
-BSUtil::mkdir_p_chown($bsdir, $BSConfig::bsuser, $BSConfig::bsgroup) || die("unable to create $bsdir\n");
-BSUtil::drop_privs_to($BSConfig::bsuser, $BSConfig::bsgroup);
-
 my $proxy;
 $proxy = $BSConfig::proxy if defined($BSConfig::proxy);
 
+sub parse_options {
+  my %opts;
+  if (!Getopt::Long::GetOptionsFromArray(\@_, \%opts,
+    'stop|exit',
+    'restart',
+    'logfile=s',
+    'dodfile=s',
+    'unparsed',
+    'pubkey=s',
+    'master=s',
+    'masterfingerprint=s',
+    'archfilter=s',
+  )) {
+    print_usage();
+    die("Invalid option(s)\n");
+  }
+  $opts{remaining} = \@_;
```

Comment:
Bah, that `remaining` is ugly. You should simply return the remaining args.

Reply:
like this ?

--- a/src/backend/bs_dodup
+++ b/src/backend/bs_dodup
@@ -64,8 +64,7 @@ sub parse_options {
     print_usage();
     die("Invalid option(s)\n");
   }
-  $opts{remaining} = \@_ if(@_);
-  return \%opts;
+  return (\%opts, @_);
 }
 
 sub print_usage {
@@ -553,7 +552,7 @@ sub daemon {
 }
 
 # copy @ARGV to keep it untouched in case of restart
-my $options = parse_options(@ARGV);
+my ($options,@args) = parse_options(@ARGV);
 
 BSUtil::mkdir_p_chown($bsdir, $BSConfig::bsuser, $BSConfig::bsgroup) || die("unable to create $bsdir\n");
 # Open logfile if requested
@@ -565,10 +564,10 @@ if ($options->{dodfile}) {
   exit(0);
 }
 
-if ($options->{remaining}) {
-  my ($prpa, $repotype, $url) = @{$options->{remaining}};
+if (@args) {
+  my ($prpa, $repotype, $url) = @args;
   my ($projid, $repoid, $arch) = split('/', ($prpa || ''), 3);
-  die("Usage: bs_dodup [--stop|--restart]\n       bs_dodup --dodfile <dodfile>\n       bs_dodup [--pubkey <pubkeyfile>] <prpa> <repotype> <url>\n") unless @{$options->{remaining}} == 3 && defined($arch);
+  die("Usage: bs_dodup [--stop|--restart]\n       bs_dodup --dodfile <dodfile>\n       bs_dodup [--pubkey <pubkeyfile>] <prpa> <repotype> <url>\n") unless @args == 3 && defined($arch);
 
   my $doddata = {
     'project' => $projid,